src/mongo_odm/model.hpp

```diff
@@ -56,13 +46,33 @@ class model {
      * @param ts
      *    The variadic pack of arguments to be forwarded to the constructor of IdType.
      */
-    template <typename... Ts, typename = std::enable_if_t<!FirstTypeIsTheSame<model, Ts...>::value>>
+    template <typename... Ts,
+              typename = std::enable_if_t<!first_two_types_are_same<model, Ts...>::value>>
     model(Ts&&... ts) : _id(std::forward<Ts>(ts)...) {
     }
 
     model() = default;
 
     /**
+     * Counts the number of documents matching the provided filter.
+     *
+     *  @param filter
+     *    The filter that documents must match in order to be counted.
+     *  @param options
+     *    Optional arguments, see mongocxx::options::count.
+     *
+     *  @return The count of the documents that matched the filter.
+     *  @throws mongocxx::exception::query if the count operation fails.
+     *
+     *  @see https://docs.mongodb.com/manual/reference/command/count/
+     */
+    static std::int64_t count(
+        bsoncxx::document::view_or_value filter,
```

Comment:
I think `count` is a safe and useful place to default this option, but the C++11 driver doesn't do that. Thoughts? I would be happy to make this:
`bsoncxx::document::view_or_value filter = bsoncxx::document::view_or_value{}`